core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -127,49 +130,50 @@ private void stopContainer(String containerId, String imageName) {
     /**
      * Register a network to be cleaned up at JVM shutdown.
      *
-     * @param networkName   the image name of the network
+     * @param id   the ID of the network
      */
-    public void registerNetworkForCleanup(String networkName) {
-        registeredNetworks.add(networkName);
+    public void registerNetworkIdForCleanup(String id) {
+        registeredNetworks.add(id);
     }
 
     /**
      * Removes any networks that contain the identifier.
-     * @param identifier
+     * @param id
      */
-    public void removeNetworks(String identifier) {
-      removeNetwork(identifier);
+    public void removeNetworkById(String id) {
+      removeNetwork(id);
     }
 
-    private void removeNetwork(String networkName) {
+    private void removeNetwork(String id) {
         try {
-            try {
```

Comment:
It could be clearer - I'll do that.
I think we still need to keep the list-with-filter operation though: if the network has already been removed for any reason, `dockerClient.removeNetworkCmd` will log out an error. So we need to do something to check for the existence of the network first that won't log an error.